### PR TITLE
Comput subrgaphs and remove duplicate nodes

### PR DESCRIPTION
**Pull request by alejoe91**
+ dump and load from folder

### CODE REVIEW

File: spikecomparison/multisortingcomparison.py [hunk 1]
```diff
@@ -10,31 +12,120 @@
...
-
+                 match_score=0.5, chance_score=0.1, n_jobs=-1, spiketrain_mode='union', verbose=False,
+                 do_matching=True):
         BaseComparison.__init__(self, sorting_list, name_list=name_list,
                                 delta_time=delta_time, sampling_frequency=sampling_frequency,
                                 match_score=match_score, chance_score=chance_score,
                                 n_jobs=n_jobs, verbose=verbose)
-        self._do_matching(verbose)
+        self._spiketrain_mode = spiketrain_mode
+        if do_matching:
```
Comment (samuelgarcia) [inline spikecomparison/multisortingcomparison.py:23]: what is the usecase of do_mathcing=False ?

File: spikecomparison/multisortingcomparison.py [hunk 2]
```diff
@@ -10,31 +12,120 @@
...
+            self._do_comparison()
+            self._do_graph()
+            self._do_agreement()
 
     def get_sorting_list(self):
+        '''
```
Comment (samuelgarcia) [inline spikecomparison/multisortingcomparison.py]: In my mind *sorting_list* was an official attribute.

File: spikecomparison/multisortingcomparison.py [hunk 3]
```diff
@@ -10,31 +12,120 @@
...
+            sorter_names = []
+            sorter_units = []
+            for node in sg.nodes:
+                sorter_names.append(node.split('_')[0])
+                sorter_units.append(int(node.split('_')[1]))
+            sg_sorter_names.append(sorter_names)
+            sg_units.append(sorter_units)
+        return sg_sorter_names, sg_units
+
+    def dump(self, save_folder):
```
Comment (samuelgarcia) [inline spikecomparison/multisortingcomparison.py:80]: Good idea, I plan to do it for other comparison for long time.
If i find soe time, we could move this at base class.

File: spikecomparison/multisortingcomparison.py (hunk 1 above)
Comment (alejoe91) [reply spikecomparison/multisortingcomparison.py:23]: when loading the graph from file

File: spikecomparison/multisortingcomparison.py (hunk 2 above)
Comment (alejoe91) [reply spikecomparison/multisortingcomparison.py]: a property? or just removing get_sorting_list?

Comment (samuelgarcia) [reply spikecomparison/multisortingcomparison.py]: keep it it is OK to have both attr and a function that return it.

File: spikecomparison/multisortingcomparison.py [hunk 4]
```diff
@@ -198,44 +227,62 @@ def _do_agreement_matrix(self, minimum_matching=0):
...
-        ax.xaxis.tick_bottom()
-        # Labels for major ticks
-        ax.set_xticklabels(sorted_name_list, fontsize=12)
-        ax.set_yticklabels(unit_ids, fontsize=12)
-
-        ax.set_xlabel('Sorters', fontsize=15)
-        ax.set_ylabel('Units', fontsize=20)
-
-        return ax
+    def _remove_duplicate_edges(self):
```
Comment (samuelgarcia) [inline spikecomparison/multisortingcomparison.py:230]: I would add a comments to explain why we have this function.
like one or two sentence.

**Comment by samuelgarcia** [comment]
I am OK with this.

**Pull request merged**